controllers/machine_controller.go

```diff
@@ -276,7 +276,10 @@ func (r *MachineReconciler) getMachinesInCluster(ctx context.Context, namespace,
 
 	machines := make([]*clusterv1.Machine, len(machineList.Items))
```

Comment:
With this change, we'll still have a slice whose length = `len(machineList.Items)`, but if we encounter a deleted Machine, we'll end up with at least one `nil` entry in the slice. I think we'll end up getting NPEs down the road in `util.GetControlPlaneMachines()`.

It's probably safer to do `var machines []*clusterv1.machine` along with `append` instead of an index-based assignment.

WDYT?

Reply:
Fixed! Great catch there